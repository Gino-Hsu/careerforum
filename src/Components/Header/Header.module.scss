@use '../../variable' as color;

.header {
  position: fixed;
  display: flex;
  align-items: center;
  width: 100%;
  height: 50px;
  background-color: #fff;
  z-index: 3;
}

.header-wrapper {
  margin: 0 auto;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 90%;
}

.avatar-container {
  padding-right: 25px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.menu-toggle {
  display: none;

  &:checked ~ .avatar-menu {
    transition: opacity 0.2s ease-out 0.15s;
    opacity: 1;
    visibility: visible;
  }
}

.avatar-menu {
  padding: 5px 0;
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 120%;
  right: 0px;
  background-color: #fff;
  border-radius: 5px;
  box-shadow: 0px 0px 2px 2px rgba(0, 0, 0, 0.1);
  opacity: 0;
  visibility: hidden;
}

.avatar-list {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100px;
  gap: 10px;
  color: color.$primary-color;
}

.avatar-item {
  padding-bottom: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
}

.header-nav {
  display: flex;
  align-items: flex-end;
  gap: 20px;
}

.header-title {
  color: color.$primary-color;
}

.nav-item {
  color: color.$primary-color;

  &:hover {
    color: color.$secondary-color;
  }
}

.header-button {
  display: flex;
  justify-content: space-between;
  gap: 1rem;
}

@media screen and (min-width: 570px) {
  .header {
    box-shadow: 1px 2px 2px 2px rgba(0, 0, 0, 0.05);
  }
}

@media screen and (min-width: 990px) {
  .header-wrapper {
    width: 80%;
  }
}
