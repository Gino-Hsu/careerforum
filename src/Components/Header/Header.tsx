import { Link } from 'react-router-dom'
import Button from '../../UIComponents/Button/Button'
import UserAvatar from '../../UIComponents/UserAvatar/UserAvatar'
import style from './Header.module.scss'

interface headerProps {
  onLoginClick: (e: React.MouseEvent) => void
  onSignUpClick: (e: React.MouseEvent) => void
  onLogoutClick: (e: React.MouseEvent) => void
  authPass: boolean
}

export default function Header(props: headerProps) {
  return (
    <header className={style['header']}>
      <div className={style['header-wrapper']}>
        <div className={style['header-nav']}>
          <Link to="/">
            <h3 className={style['header-title']}>Career Forum</h3>
          </Link>
          <div className={style['nav-link']}>
            <Link to="/careerforum/home">
              <nav className={style['nav-item']}>問答版</nav>
            </Link>
          </div>
        </div>
        <div className={style['header-button']}>
          {!props.authPass && (
            <>
              <Button
                type="button"
                innerText="登入"
                style="button-login"
                onClick={props.onLoginClick}
                disabled={false}
              />
              <Button
                type="button"
                innerText="註冊"
                style="button-sign-up"
                onClick={props.onSignUpClick}
                disabled={false}
              />
            </>
          )}
          {props.authPass && (
            <UserAvatar
              userAvatar={currentUser.avatar}
              avatarStyle={'header-user-avatar'}
            />
          )}
        </div>
      </div>
    </header>
  )
}

const currentUser = {
  id: '1', // user PK:id
  role: 'TA',
  name: '', // 可填可不填
  account: 'user1',
  email: 'user1@careerForum.com',
  avatar: 'https://cdn-icons-png.flaticon.com/512/1864/1864514.png',
  cover: 'http://...',
  createdAt: '2023/01/07',
  updatedAt: '2023/01/07',
}
