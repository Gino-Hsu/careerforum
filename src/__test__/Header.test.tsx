import React from 'react'
import { render, screen } from '@testing-library/react'
import Header from '../Components/Header/Header'
import { BrowserRouter } from 'react-router-dom'

test('Should display the "登入" and "註冊" buttons when authPass is false.', () => {
  const handleClick = jest.fn()

  render(
    <BrowserRouter>
      <Header
        onLoginClick={handleClick}
        onSignUpClick={handleClick}
        onLogoutClick={handleClick}
        authPass={false}
      />
    </BrowserRouter>
  )

  const loginElement = screen.getByText(/登入/i)
  const signUpElement = screen.getByText(/註冊/i)

  expect(loginElement).toBeInTheDocument
  expect(signUpElement).toBeInTheDocument
})

test('Should display the "登出" button when authPass is true.', () => {
  const handleClick = jest.fn()

  render(
    <BrowserRouter>
      <Header
        onLoginClick={handleClick}
        onSignUpClick={handleClick}
        onLogoutClick={handleClick}
        authPass={true}
      />
    </BrowserRouter>
  )
  const headerUserAvatar = screen.getByRole('img', {
    name: /使用者頭像/i
  })

  expect(headerUserAvatar).toBeInTheDocument
})

test('Should not display the "註冊" button when authPass is true.', () => {
  const handleClick = jest.fn()
  render(
    <BrowserRouter>
      <Header
        onLoginClick={handleClick}
        onSignUpClick={handleClick}
        onLogoutClick={handleClick}
        authPass={false}
      />
    </BrowserRouter>
  )

  const loginElement = screen.getByText(/登入/i)
  const signUpElement = screen.queryByText(/註冊/i)

  expect(loginElement).toBeInTheDocument
  expect(signUpElement).not.toBeInTheDocument
})
