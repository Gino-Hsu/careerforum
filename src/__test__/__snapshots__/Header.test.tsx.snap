// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be show signUp Modal on press of Header signUp button 1`] = `
<div
  id="modal-root"
>
  <div
    class="modal-outer"
  >
    <div
      class="backdrop"
    />
    <div
      class="modal-container scrollbar"
    >
      <button
        class="button-close"
        type="button"
      >
        <p
          class="button-text"
        >
          X
        </p>
      </button>
      <header
        class="header"
      >
        <h2>
          註冊
        </h2>
      </header>
      <form
        class="form"
      >
        <div
          class="input-container"
        >
          <input
            class="input"
            id="email"
            name="email"
            placeholder=" "
            required=""
            type="email"
            value="email"
          />
          <span
            class="error-message"
          />
          <label
            class="label"
            for="email"
          >
            Email
          </label>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Should be show signUp Modal with onClick Header signUp button 1`] = `
<div
  id="modal-root"
>
  <div
    class="modal-outer"
  >
    <div
      class="backdrop"
    />
    <div
      class="modal-container scrollbar"
    >
      <button
        class="button-close"
        type="button"
      >
        <p
          class="button-text"
        >
          X
        </p>
      </button>
      <header
        class="header"
      >
        <h2>
          註冊
        </h2>
      </header>
      <form
        class="form"
      >
        <div
          class="input-container"
        >
          <input
            class="input"
            id="email"
            name="email"
            placeholder=" "
            required=""
            type="email"
            value="email"
          />
          <span
            class="error-message"
          />
          <label
            class="label"
            for="email"
          >
            Email
          </label>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`show signUp Modal with onClick Header signUp button 1`] = `
<div
  id="modal-root"
>
  <div
    class="modal-outer"
  >
    <div
      class="backdrop"
    />
    <div
      class="modal-container scrollbar"
    >
      <button
        class="button-close"
        type="button"
      >
        <p
          class="button-text"
        >
          X
        </p>
      </button>
      <header
        class="header"
      >
        <h2>
          註冊
        </h2>
      </header>
      <form
        class="form"
      >
        <div
          class="input-container"
        >
          <input
            class="input"
            id="email"
            name="email"
            placeholder=" "
            required=""
            type="email"
            value="email"
          />
          <span
            class="error-message"
          />
          <label
            class="label"
            for="email"
          >
            Email
          </label>
        </div>
      </form>
    </div>
  </div>
</div>
`;
